lib/facter/consul_version.rb

```diff
@@ -4,7 +4,12 @@
   confine :kernel => 'Linux'
   setcode do
     begin
-      Facter::Util::Resolution.exec('consul --version 2> /dev/null').lines.first.split[1].tr('v','')
+      tmp = ENV['PATH']
+      path = ENV.fetch('PATH') { '/bin:/usr/bin:/usr/local/bin' }
+      ENV['PATH'] = path + ':/usr/local/bin' unless path =~ /(:|^)\/usr\/local\/bin\/?(:|$)/
```

Comment:
I think this is overkill. It doesn't hurt to append the path even if it is already there.
Just save the path, append the path, exec, reset path